### PR TITLE
chore: add CodeQL Action

### PR DESCRIPTION
**Pull request by XhmikosR**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Should help with the security issues in the future.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

---

Feel free to adapt the `on` event or schedule to match your needs.

### CODE REVIEW

**Comment by XhmikosR** [comment]
To clarify in case you are not aware, you can ignore specific errors/warnings that are intentional (like used in tests) in the security tab.

CodeQL is not a 1-to-1 replacement to [LGTM](https://lgtm.com/projects/g/netlify/cli/?mode=list), but at least it covers the security-related issues.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: 'CodeQL'
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
```
Comment (erezrokah) [inline .github/workflows/codeql.yml:10]: Do we need the `schedule` and `workflow_dispatch`?

What is the use cases for them?

**Review by erezrokah — CHANGES_REQUESTED** [review]
This is @XhmikosR, wondering if we need the schedule and dispatch triggers

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (XhmikosR) [reply .github/workflows/codeql.yml:10]: Well, the schedule is what it literally means: run the workflow on a schedule. You can adapt this or remove it.

`workflow_dispatch` allows to run the workflow on demand, it was recently added on Actions as a feature.

Comment (erezrokah) [reply .github/workflows/codeql.yml:10]: Sorry, let me clarify - do they provide added value assuming we run this for each PR and push to `main`?

**Review by erezrokah — APPROVED** [review]
🚀

**Pull request merged**